fe/fe-core/src/main/java/com/starrocks/replication/LakeReplicationJob.java

```diff
@@ -0,0 +1,160 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.replication;
+
+import com.google.gson.annotations.SerializedName;
+import com.staros.client.StarClientException;
+import com.staros.proto.FileCacheInfo;
+import com.staros.proto.FilePathInfo;
+import com.staros.proto.ShardInfo;
+import com.starrocks.common.MetaNotFoundException;
+import com.starrocks.lake.StarOSAgent;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.server.StorageVolumeMgr;
+import com.starrocks.server.WarehouseManager;
+import com.starrocks.storagevolume.StorageVolume;
+import com.starrocks.task.ReplicateSnapshotTask;
+import com.starrocks.thrift.TTableReplicationRequest;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class LakeReplicationJob extends ReplicationJob {
+    private static final Logger LOG = LogManager.getLogger(LakeReplicationJob.class);
+    private final long fakedShardId;
+    static Map<String, Long> storageVolumeNameToShardIdMap = new ConcurrentHashMap<>();
+
+    @SerializedName(value = "srcDatabaseId")
+    protected final long srcDatabaseId;
+
+    @SerializedName(value = "srcTableId")
+    protected final long srcTableId;
+
+    public LakeReplicationJob(TTableReplicationRequest request) throws MetaNotFoundException {
+        super(request);
+        this.srcDatabaseId = request.src_database_id;
+        this.srcTableId = request.src_table_id;
+        this.fakedShardId = getFakedShardId(request.src_storage_volume_name, request.src_service_id);
+    }
+
+    private synchronized long getFakedShardId(String svName, String srcServiceId) {
+        return storageVolumeNameToShardIdMap.computeIfAbsent(svName,
+                storageVolumeName -> getOrCreateFakeShard(storageVolumeName, srcServiceId));
+    }
+
+    private long getOrCreateFakeShard(String storageVolumeName, String srcServiceId) {
```

Comment:
replace "FakeShard" concept with "tablet", `Shard` should be only in StarOS, turn to virtual tablet or tmp tablet or whatever if the logic is implemented inside Frontend.